service/src/test/java/uk/gov/hmcts/reform/fpl/service/orders/generator/C29RecoveryOfAChildDocumentParameterGeneratorTest.java

```diff
@@ -0,0 +1,304 @@
+package uk.gov.hmcts.reform.fpl.service.orders.generator;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.mockito.junit.jupiter.MockitoSettings;
+import org.mockito.quality.Strictness;
+import uk.gov.hmcts.reform.fpl.config.LocalAuthorityNameLookupConfiguration;
+import uk.gov.hmcts.reform.fpl.enums.C29ActionsPermitted;
+import uk.gov.hmcts.reform.fpl.enums.PlacedUnderOrder;
+import uk.gov.hmcts.reform.fpl.model.Address;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+import uk.gov.hmcts.reform.fpl.model.Child;
+import uk.gov.hmcts.reform.fpl.model.common.Element;
+import uk.gov.hmcts.reform.fpl.model.event.ManageOrdersEventData;
+import uk.gov.hmcts.reform.fpl.service.ChildrenService;
+import uk.gov.hmcts.reform.fpl.service.orders.docmosis.C29RecoveryOfAChildDocmosisParameters;
+import uk.gov.hmcts.reform.fpl.service.orders.docmosis.C29RecoveryOfAChildDocmosisParameters.C29RecoveryOfAChildDocmosisParametersBuilder;
+import uk.gov.hmcts.reform.fpl.service.orders.docmosis.DocmosisParameters;
+
+import java.time.LocalDate;
+import java.util.List;
+
+import static java.lang.String.format;
+import static java.util.Objects.isNull;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.mockito.Mockito.when;
+import static uk.gov.hmcts.reform.fpl.enums.DocmosisTemplates.ORDER_V2;
+import static uk.gov.hmcts.reform.fpl.model.order.Order.C29_RECOVERY_OF_A_CHILD;
+import static uk.gov.hmcts.reform.fpl.utils.DateFormatterHelper.DATE_WITH_ORDINAL_SUFFIX;
+import static uk.gov.hmcts.reform.fpl.utils.DateFormatterHelper.formatLocalDateToString;
+import static uk.gov.hmcts.reform.fpl.utils.DateFormatterHelper.getDayOfMonthSuffix;
+import static uk.gov.hmcts.reform.fpl.utils.ElementUtils.wrapElements;
+
+@ExtendWith({MockitoExtension.class})
+@MockitoSettings(strictness = Strictness.LENIENT)
+class C29RecoveryOfAChildDocumentParameterGeneratorTest {
+
+    private static final String LA_CODE = "LA_CODE";
+    private static final String LA_NAME = "Local Authority Name";
+    private static final String FURTHER_DIRECTIONS = "further directions";
+    private static final String OFFICER_NAME = "Officer Barbrady";
+    private static final LocalDate ORDER_CREATED_DATE = LocalDate.of(2020, 8, 20);
+    private static final String dayOrdinalSuffix = getDayOfMonthSuffix(ORDER_CREATED_DATE.getDayOfMonth());
```

Comment:
Two comments:

1. If you're going to use this constant, you should name it as a constant.
2. Maybe instead of using logic to instantiate this constant, you could just put the value directly. It will be easier to read.

Reply:
Very good points. Ill update now